### PR TITLE
chore(NA): provides Apple Silicon ARM build

### PR DESCRIPTION
**Pull request by mistic**
Closes https://github.com/bazelbuild/bazel-watcher/issues/493
Closes https://github.com/bazelbuild/bazel-watcher/issues/482

This PR introduces the necessary changes in order to provide `ibazel` ARM builds for Apple Silicon. 
Along the way it also upgrades the go version to `1.17.6` as it was needed as part of the process.

`bazel-watcher` seems to be the last piece of the Bazel ecosystem missing `darwin_arm64` support so I believe this PR is of high priority.

I've cross compiled this locally on an Intel Mac and run the binary to confirm it works as expected in an ARM Mac.

### CODE REVIEW

File: bazel/bazel_test.go [hunk 1]
```diff
@@ -94,26 +94,31 @@ func TestCancel(t *testing.T) {
...
+	}
+	if err := os.MkdirAll(bazelNpmDir+"/bazel-windows_x64", 0755); err != nil {
 		t.Errorf(err.Error())
 	}
-	if err := os.MkdirAll(bazelNpmDir + "/bazel-windows_x64", 0755); err != nil {
+	if err := os.MkdirAll(bazelNpmDir+"/bazel-darwin_x64", 0755); err != nil {
 		t.Errorf(err.Error())
 	}
-	if err := os.MkdirAll(bazelNpmDir + "/bazel-darwin_x64", 0755); err != nil {
+	if err := os.MkdirAll(bazelNpmDir+"/bazel-darwin_arm64", 0755); err != nil {
```
Comment (achew22) [inline bazel/bazel_test.go:121]: Please run `gofmt` on this file

Comment (mistic) [reply bazel/bazel_test.go:121]: I've run `gofmt -s -w ./bazel/bazel_test.go` but it does not change anything on the file

**Comment by mistic** [comment]
I've also reverted the change at https://github.com/bazelbuild/bazel-watcher/pull/496/commits/e7bb40f7d5afabd9d79eaaa181306916c8e20916 but just an heads up the CI BUILD don't go through without it and complains about a `missing stamp`. As I'm not very familiar with the release process of that project will let to you to decide what to do here

**Comment by mistic** [comment]
@achew22 do you think we can move on with this PR (and maybe, if you have the time for it, my other PRs as well https://github.com/bazelbuild/bazel-watcher/pull/494 https://github.com/bazelbuild/bazel-watcher/pull/479)?

File: release/npm/generate_package.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ type jsPackage struct {
 // See:
 //   * https://github.com/bazelbuild/rules_go/issues/2379
 //   * https://github.com/golang/go/issues/37369
-var Version string = "VERSION_NOT_SET"
+var Version string = "0.16.0"
```
Comment (achew22) [inline release/npm/generate_package.go:25]: Manually changing this means that the stamping logic (`x_defs`) is broken which means that the releasing code will not work. Can you figure out what's going on here? I'd rather not move back to manually releasing

Comment (mistic) [reply release/npm/generate_package.go:25]: What are the reasons that could make that stamping logic to fail? I'm not familiar with what you are doing there but I dont have a reason on top of my head  around what I've changed that could make this to fail

Comment (mistic) [reply release/npm/generate_package.go]: I've found a solution for this.

**Comment by mistic** [comment]
@achew22 CI is now passing without breaking the stamping logic

File: .bazelrc [hunk 3]
```diff
@@ -1,5 +1 @@
-build --workspace_status_command="python tools/workplace_status.py"
-
-# Flags used during releases
-build:release --stamp
-test:release --stamp
+build --workspace_status_command="python tools/workplace_status.py" --stamp
```
Comment (achew22) [inline .bazelrc:1]: Can you restore the config mode of `release`? Forcing stamp on all the time causes cache misses when I checked this out locally for testing.

**Review by achew22 — COMMENTED** [review]
One minor comment then we should be good to go

File: .bazelrc (hunk 3 above)
Comment (mistic) [reply .bazelrc]: Alright, done!

**Comment by mistic** [comment]
@achew22 I think this PR is ready to go

File: release/npm/BUILD [hunk 4]
```diff
@@ -18,16 +18,25 @@ genrule(
     name = "package",
     srcs = [
         "//:CONTRIBUTORS",
+        ":npm"
     ],
     outs = ["package.json"],
-    cmd = "$(location :npm) $(location //:CONTRIBUTORS) > $@",
+    cmd = "$(location :package_sh) ./$(location :npm) $(location //:CONTRIBUTORS) > $@",
```
Comment (achew22) [inline release/npm/BUILD:24]: Why do we need this indirection? What happens if you go back to the direct invocation of `:npm` from the genrule?

Comment (mistic) [reply release/npm/BUILD:24]: The indirection is a workaround for a problem I found when upgrading into the latest `rules_go@0.30.0` however I was able to reproduce and verify the problem as early as upgrading into [v0.23.0](https://github.com/bazelbuild/rules_go/releases/tag/v0.23.0)

Below you can find the error output:

```
bazel-watcher tiagocosta$ bazel build --config=release //release/npm:package
INFO: Analyzed target //release/npm:package (1 packages loaded, 36 targets configured).
INFO: Found 1 target...
ERROR: /projects/bazel-watcher/release/npm/BUILD:17:8: Executing genrule //release/npm:package failed: (Exit 2): bash failed: error executing command /bin/bash -c 'source external/bazel_tools/tools/genrule/genrule-setup.sh; bazel-out/host/bin/release/npm/npm_/npm ./CONTRIBUTORS > bazel-out/darwin-fastbuild/bin/release/npm/package.json'

Use --sandbox_debug to see verbose messages from the sandbox
panic: The version string was not overriden. Please rebuild with --stamp

goroutine 1 [running]:
main.main()
	release/npm/generate_package.go:29 +0x3fc
Target //release/npm:package failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 2.707s, Critical Path: 1.57s
INFO: 6 processes: 4 internal, 2 darwin-sandbox.
FAILED: Build did NOT complete successfully
```

I really think it is related with that was opening by you at https://github.com/bazelbuild/rules_go/issues/2224 that was not reproducible somewhere in between `v0.19.4` and `v0.22.4` but is now a problem again. For the sake of what we are doing here I think we can live with the indirection.

Comment (achew22) [reply release/npm/BUILD:24]: It's probably transition related, those things are beasts. Okay, let's merge this and get a release out there for all the M1 owners. Thanks for fighting through this!

**Comment by jonrose-dev** [comment]
Very excited for this! Is there estimated time line for when this is expected to be released?

**Comment by mistic** [comment]
I have a fix for it @jonrose-dev https://github.com/bazelbuild/bazel-watcher/pull/510

**Pull request merged**